dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchCollection.cs

```
@@ -125,53 +125,77 @@ public override async Task<bool> CollectionExistsAsync(CancellationToken cancell
     }
 
     /// <inheritdoc />
-    public override Task CreateCollectionAsync(CancellationToken cancellationToken = default)
+    public override async Task EnsureCollectionExistsAsync(CancellationToken cancellationToken = default)
     {
-        var vectorSearchConfig = new VectorSearch();
-        var searchFields = new List<SearchField>();
+        const string OperationName = "CreateIndex";
 
-        // Loop through all properties and create the search fields.
-        foreach (var property in this._model.Properties)
+        // Don't even try to create if the collection already exists.
+        if (await this.CollectionExistsAsync(cancellationToken).ConfigureAwait(false))
```

Not a problem, but what's the purpose of the precheck? The only reason I see for it is maybe to avoid errors showing up in the database logs for trying to create a collection that already exists? Pointing this out since it does an extra roundtrip when the collection doesn't exist, and we could just optimistically call Create below and ignore the error instead.

(also applies to the other providers)

---

Yeah, it's to avoid additional errors in the logs.  I only did it for cases where we rely on catching an exception to determine whether it already exists.  I think the most common scenario will be that it exists already, so the extra roundtrip is probably ok in the case that it doesn't.

---

OK, no issue from my side.